crates/bevy_asset/src/server/mod.rs

```diff
@@ -462,12 +491,20 @@ impl AssetServer {
     /// [`RecursiveDependencyLoadState`].
     #[must_use = "not using the returned strong handle may result in the unexpected release of the assets"]
     pub fn load_folder<'a>(&self, path: impl Into<AssetPath<'a>>) -> Handle<LoadedFolder> {
```

Comment:
[Some feedback from Cart in Discord](https://discord.com/channels/691052431525675048/749332104487108618/1164691159445147809) was that this needs support for hot reloading since folder assets can no longer be manually re-loaded. This means that file change events need to be recursively applied to folders that the files are contained in.

Reply:
Folders now reload when any file in them or in sub directories changes. They do not reload when a folder is created/removed/renamed, since sub-folders are irrelevant by themselves and any file contained in them triggers a separate asset event that will cause a reload.